### PR TITLE
[HttpFoundation] move ini calls to separate functions in PDO session …

### PR DESCRIPTION
**Pull request by gseidel**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

Make separate functions for ini calls in PdoSessionHandler. So it's easier to extend this class and change the parameters without changing the ini config.

### CODE REVIEW

**Comment by Tobion** [comment]
What speaks against using the ini config that are made for this?

**Comment by gseidel** [comment]
The ini config made for this, but if you want to configure it e.g. in a config.yml it's not easy to pass that value to the PdoSessionHandler except overwriting the ini by ini_set. I think it's a nicer solution if you can pass the values over the constructor, i want to do this in my project and extend the PdoSessionHandler class for this, but for the maxlifetime i have to copy a lot of code. With this PR it would make it easier for me. Makes that sense for you or would you prefer a different solution?

**Comment by Tobion** [comment]
Using the ini allows to configure different lifetimes per session. As a constructor injection is inferior to this and doesn't solve a real problem, I do not see the point of adding more extension points.

File: src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php [hunk 1]
```diff
@@ -687,9 +697,19 @@ private function getMergeSql()
     protected function getConnection()
     {
         if (null === $this->pdo) {
-            $this->connect($this->dsn ?: ini_get('session.save_path'));
+            $this->connect($this->dsn ?: $this->getSessionSavePath());
```
Comment (stof) [inline src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php]: -1 for this. there is already a way to configure the DSN without using the ini variable

Comment (gseidel) [inline src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php]: You are correct i will remove these lines.

**Comment by gseidel** [comment]
@Tobion I use a different approach. You can configure the maxlifetime over a constructor option. If no one is set, then the ini setting is used as before. I'm still thinking that someone want to define the maxliftime without touching the ini config.

**Comment by Tobion** [comment]
Then just set the ini in your custom constructor. Then it's not visible to the outside. Et voila

File: src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php [hunk 2]
```diff
@@ -321,8 +328,6 @@ public function destroy($sessionId)
      */
     public function write($sessionId, $data)
     {
-        $maxlifetime = (int) ini_get('session.gc_maxlifetime');
```
Comment (Tobion) [inline src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php]: this removes the possibily to set a lifetime per session as I explained. So this is a BC break.

**Comment by fabpot** [comment]
I tend to agree with @Tobion.

**Pull request closed without merging**